plugin/core/sessions.py

```diff
@@ -387,10 +385,14 @@ def _handle_request_workspace_configuration(self, params: Dict[str, Any], reques
     def _handle_register_capability(self, params: Any, request_id: Any) -> None:
         registrations = params["registrations"]
         for registration in registrations:
-            capability_path, registration_path = method_to_capability(registration["method"])
+            method = registration["method"]
+            capability_path, registration_path = method_to_capability(method)
             debug("{}: registering capability:".format(self.config.name), capability_path)
             set_dotted_value(self.capabilities, capability_path, registration["registerOptions"])
             set_dotted_value(self.capabilities, registration_path, registration["id"])
+            if method == "workspace/didChangeConfiguration":
+                # The mega-exceptional case.
+                self.client.send_notification(Notification.didChangeConfiguration({'settings': self.config.settings}))
```

Comment:
I think it might be wrong to send it before sending a response to registration.
The server might only register a handler for it after it receives the response to registration.

Reply:
BTW. VSCode doesn't send it in this case. Even though eslint registers this notification, vscode doesn't send it at all in the session (without changing VSCode settings at least).

Reply:
Fixed in https://github.com/sublimelsp/LSP/pull/1005/commits/c087319b7ef27599cc6de01bec31317587b5ad50

Reply:
> BTW. VSCode doesn't send it in this case. Even though eslint registers this notification, vscode doesn't send it at all in the session (without changing VSCode settings at least).

If we apply the same behavior, then we never actually send `didChangeConfiguration`. This is because settings changes from LSP.sublime-settings, view settings, project settings are not propagated into the plugin.

When we get to a point that we do propagate settings changes, we can perhaps change the above behavior to only notify server when settings actually change.

Reply:
I'm fine with current behavior but chances are we don't need to send it then.